pkg/orchestration/wiring/datadog/plugin.go

```diff
@@ -97,10 +103,10 @@ func constructServiceInstance(resource *orch_v1.StateResource, context *wiringpl
 	}
 
 	alarmInstanceResource := smith_v1.Resource{
-		Name: createAlarmNamesForSmithResource(resource.Name, deploymentResourceName, alarmType),
+		Name: wiringutil.ConsumerProducerResourceNameWithPostfix(resource.Name, kubeCompute, string(alarmType)),
```

Comment:
I'm using pre-existing helper functions here - this autowiring function should probably be changed to allow depending on multiple computes (why not?) for them to get the same alarms. And then it becomes a necessity to include the compute name into the resource name here (the "producer") to avoid clashes. Can be done separately.